includes/optional-modules/class-woo-member-commenting.php

```diff
@@ -0,0 +1,163 @@
+<?php
+/**
+ * Woo member commenting.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Woo member commenting.
+ */
+class Woo_Member_Commenting {
+
+	/**
+	 * Initialize.
+	 *
+	 * @return void
+	 */
+	public static function init(): void {
+		add_action( 'init', [ __CLASS__, 'on_init' ] );
+	}
+
+	/**
+	 * Action callback on init.
+	 *
+	 * @return void
+	 */
+	public static function on_init() {
+		if ( ! self::should_load() ) {
+			return;
+		}
+
+		add_filter( 'comment_form_defaults', [ __CLASS__, 'filter_comment_form_defaults' ] );
+		add_filter( 'comment_form_fields', [ __CLASS__, 'filter_comment_form_fields' ] );
+		add_filter( 'comment_form_submit_field', [ __CLASS__, 'filter_comment_form_submit_field' ] );
+	}
+
+	/**
+	 * Filter callback.
+	 *
+	 * @param array $defaults The defaults array to filter.
+	 *
+	 * @return mixed
+	 */
+	public static function filter_comment_form_defaults( $defaults ) {
+		if ( ! self::require_membership_to_comment_for_user() ) {
+			return $defaults;
+		}
+
+		$defaults['comment_notes_before'] = '';
+		$defaults['must_log_in']          = sprintf( '<p class="must-log-in">%s</p>', self::get_membership_required_message() );
+
+		return $defaults;
+	}
+
+
+	/**
+	 * Filter callback.
+	 *
+	 * @param array $fields The fields array to filter.
+	 *
+	 * @return array The filtered fields array.
+	 */
+	public static function filter_comment_form_fields( $fields ) {
+		return self::require_membership_to_comment_for_user() ? [] : $fields;
+	}
+
+	/**
+	 *  Filter callback.
+	 *
+	 * @param string $field Field to filter.
+	 *
+	 * @return string The filtered field.
+	 */
+	public static function filter_comment_form_submit_field( $field ) {
+		if ( ! self::require_membership_to_comment_for_user() ) {
+			return $field;
+		}
+
+		return sprintf( '<p class="must-log-in">%s</p>', self::get_membership_required_message() );
+	}
+
+	/**
+	 * Get the message to display to users that don't have access to commenting.
+	 *
+	 * @return string The message to display.
+	 */
+	private static function get_membership_required_message(): string {
+		$message = __( 'Only Members may post a comment.', 'newspack-plugin' );
+		if ( ! is_user_logged_in() ) {
+			$message .= ' ' . __( 'If you already have a membership, then <a href="/my-account/">sign in</a>.', 'newspack-plugin' );
```

Comment:
It's possible that some sites may have changed the base slug for the My Account page. Something like this could help avoid hard-coding a wrong URL for those sites:

```suggestion
			$message .= ' ' . sprintf( __( 'If you already have a membership, then <a href="%s">sign in</a>.', 'newspack-plugin' ), function_exists( 'wc_get_account_endpoint_url' ) ? \wc_get_account_endpoint_url( 'edit-account' ) : home_url( 'my-account' ) );
```

Reply:
Good call! 👍